common/src/main/java/dev/latvian/mods/kubejs/item/custom/MultitoolItemJS.java

```diff
@@ -0,0 +1,179 @@
+            speedBaseline = speed.add(speed.multiply(BigDecimal.valueOf(.05)))
+                .setScale(1, CEILING).floatValue();
+        }
+        @Override
+        public Item createObject() {
+            setValues();
+            return new MultitoolItemJS(attackDamageBaseline, speedBaseline, toolTier, createItemProperties(), this);
+        }
+    }
+    public final Set<TagKey<Block>> mineableTags;
+    protected boolean isAxe, isHoe, isPickaxe, isShovel;
+    private final Multimap<ResourceLocation, AttributeModifier> attributes;
+    private boolean modified = false;
+    public MultitoolItemJS(float attack, float speed, MutableToolTier tier, Properties properties, Builder builder) {
+        super(attack, speed, tier, null, properties);
+        defaultModifiers = ArrayListMultimap.create(defaultModifiers);
+        mineableTags = Set.copyOf(builder.mineableTags);
+        isAxe = builder.isAxe;
+        isHoe = builder.isHoe;
+        isPickaxe = builder.isPickaxe;
+        isShovel = builder.isShovel;
+        attributes = builder.attributes;
+    }
+    public boolean isAxe() {
+        return isAxe;
+    }
+    public boolean isHoe() {
+        return isHoe;
+    }
+    public boolean isPickaxe() {
+        return isPickaxe;
+    }
+    public boolean isShovel() {
+        return isShovel;
+    }
+    @Override
+    public float getDestroySpeed(ItemStack itemStack, BlockState blockState) {
+        return isInMineables(blockState) ? speed : 1f;
+    }
+    @Override
+    public Multimap<Attribute, AttributeModifier> getDefaultAttributeModifiers(EquipmentSlot equipmentSlot) {
+        if (!modified) {
+            modified = true;
+            attributes.forEach((r, m) -> defaultModifiers.put(KubeJSRegistries.attributes().get(r), m));
+        }
+        return super.getDefaultAttributeModifiers(equipmentSlot);
+    }
+    @Override
+    // right-clicking on dirt will till, unless sneaking, then it will turn it into a path
+    public InteractionResult useOn(UseOnContext ctx) {
+        return (isAxe && Items.IRON_AXE.useOn(ctx) != InteractionResult.PASS) ||
+            (isShovel && (!isHoe || Optional.ofNullable(ctx.getPlayer())
+                    .map(Entity::isCrouching).orElse(false)) && // player sneaking check
+                Items.IRON_SHOVEL.useOn(ctx) != InteractionResult.PASS) ||
+            (isHoe && Items.IRON_HOE.useOn(ctx) != InteractionResult.PASS) ||
+            (isPickaxe && Items.IRON_HOE.useOn(ctx) != InteractionResult.PASS) ?
+            InteractionResult.sidedSuccess(ctx.getLevel().isClientSide) : InteractionResult.PASS;
+    }
+    @Override
+    public boolean isCorrectToolForDrops(BlockState state) {
+        final int i = tier.getLevel();
+        if ((i < 3 && state.is(BlockTags.NEEDS_DIAMOND_TOOL)) ||
+            (i < 2 && state.is(BlockTags.NEEDS_IRON_TOOL)) ||
+            (i < 1 && state.is(BlockTags.NEEDS_STONE_TOOL))) return false;
```

Comment:
Mods can add more tiers, even Forge by default adds gold and netherite as ones.
This should really handle those as well. not be hardcoded to vanilla.

Reply:
To be fair, this is a larger issue with KubeJS as a whole, since harvest levels are... pretty terribly done across Vanilla, Fabric and Forge. I'll need to find an abstraction for it, maybe even add it to Architectury

Reply:
ive done a fix on forge, but have no idea how to do this on fabric

Reply:
Fabric has number based tiers for everything thats not vanilla iirc, so you can just append the passed in number to a string to get the tag if its not a default one. Their wiki probably has a page on it.